### DISCUSSION TITLE
Help simplify custom postgresql date_trunc function

### DISCUSSION THREAD

**Discussion by emorozov** [tortoise/tortoise-orm#1390] [Q&A]
The only way I've found after day of trial and errors:

```
from pypika import CustomFunction
from tortoise.expressions import F, Function, RawSQL

class DateTrunc(Function):
    database_func = CustomFunction('DATE_TRUNC', ["day", "modified_at"])

await Vote.annotate(day=DateTrunc(RawSQL("'day'"), F('modified_at'))).filter(
    day__gt=date.today()).distinct().values_list('object_id', 'object_type_id')
```

This looks ad hoc and too cumbersome, I'm sure there's a nicer way to avoid explicit `RawSQL` and `F` expressions, but documentation contains only trivial example and I couldn't come up with anything better than this.

**Comment by long2ice** [comment]
Maybe we can provide a shortcut like sqlalchemy? Just like `func.sum`. But how type checking for that?
